File: App/app/src/main/java/com/github/travelplannerapp/login/LoginPresenter.java
package com.github.travelplannerapp.login;

import com.github.travelplannerapp.BasePresenter;

public class LoginPresenter extends BasePresenter<LoginContract.View>
        implements LoginContract.Presenter {

    protected LoginPresenter(LoginContract.View view) {
        super(view);
    }

    @Override
    public void signIn() {
        //example of the view and presenter flow
        view.showTravels();
    }
}

File: App/app/src/main/java/com/github/travelplannerapp/login/LoginPresenter.kt
package com.github.travelplannerapp.login

import com.github.travelplannerapp.BasePresenter

class LoginPresenter(view: LoginContract.View) : BasePresenter<LoginContract.View>(view), LoginContract.Presenter {

    override fun signIn() {
        //example of the view and presenter flow
        view.showTravels()
    }
}
